lib/iris/fileformats/grib/message.py

```diff
@@ -481,9 +481,24 @@ def get_computed_key(self, key):
         if key in vector_keys:
             res = gribapi.grib_get_array(self._message_id, key)
         else:
-            res = gribapi.grib_get(self._message_id, key)
+            res = self._get_value_or_missing(key)
         return res
 
     def keys(self):
         """Return coded keys available in this Section."""
         return self._keys
+
+    def _get_value_or_missing(self, key, use_int=False):
```

Comment:
👍 for having factored this out
Rather than a binary toggle, would you mind passing the dtype of the expected result?